tsdb/store.go

```diff
@@ -373,41 +373,64 @@ func (s *Store) ShardIteratorCreator(id uint64) influxql.IteratorCreator {
 	return &shardIteratorCreator{sh: sh}
 }
 
+// shardExecuteFn checks if a function should be run on a shard and returns
+// a function to execute on that shard if it should.
+type shardExecuteFn func(shardID uint64, sh *Shard) func() error
```

Comment:
Returning a `func() error` doesn't seem necessary?  Why can it not just return an `error`?